### PR TITLE
Add event store timeline query API endpoints

### LINKED ISSUE #737: 6.2.1: Add event store timeline query API endpoints
## Summary

Create backend API endpoints for browsing domain events chronologically with filtering and cursor-based pagination.

## Endpoints

- `GET /api/timeline/events` — paginated event list from `$all` stream
  - Params: `cursor` (global_position), `limit` (default 50, max 200), `order` (asc/desc), `stream_category`, `event_type`, `aggregate_id`, `kind` (EVENT/COMMAND)
- `GET /api/timeline/events/{message_id}` — single event detail (full payload + metadata)
- `GET /api/timeline/stats` — summary stats (total events, events/minute, active streams, last event timestamp)

## Key Files

- Create: `src/protean/server/observatory/routes/timeline.py`
- Modify: `src/protean/server/observatory/routes/__init__.py` (wire `create_timeline_router`)
- Create: `tests/server/observatory/test_observatory_timeline_api.py`

## Implementation Notes

- Follow the `routes/eventstore.py` pattern: standalone data-collection functions + router factory
- Pagination via global_position cursor (last seen position → next page)
- Filtering applied in Python after reading chunks from event store (no port changes needed)
- Multi-domain: iterate domains to aggregate results, tag each event with domain name
- Graceful degradation when event store unavailable
- Instrument endpoints with OTEL spans (from 6.1)

Part of epic #736

### PR DESCRIPTION
**Pull request by subhashb**
## Summary

- Add three new Observatory API endpoints for chronological event browsing:
  - `GET /api/timeline/events` — paginated event list from `$all` stream with cursor-based pagination and filtering (stream_category, event_type, aggregate_id, kind)
  - `GET /api/timeline/events/{message_id}` — single event detail with full payload and metadata
  - `GET /api/timeline/stats` — summary statistics (total events, active streams, events/minute, last event timestamp)
- Follow established Observatory route pattern: standalone data-collection functions + router factory with graceful degradation
- Wire timeline router into Observatory via `routes/__init__.py`

## Test plan

- [x] 51 tests covering all functions, endpoints, edge cases, and graceful degradation
- [x] Unit tests for serialization helpers and metadata extraction functions
- [x] Integration tests for data-collection functions with real in-memory event store
- [x] HTTP endpoint tests verifying response shapes, pagination, filtering, and input validation
- [x] Route wiring verification
- [x] Standalone router test with mock domain (no infrastructure needed)
- [x] ~90% code coverage on new module

Closes #737

### CODE REVIEW

File: src/protean/server/observatory/routes/timeline.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                for msg in raw_messages:
+                    all_events.append((msg, domain.name))
+        except Exception:
+            logger.debug(
+                "Failed to read events from %s", domain.name, exc_info=True
+            )
+
+    # Sort by global_position (ascending)
+    all_events.sort(key=lambda x: x[0].get("global_position", 0))
+
```
[the remark this answers is not in the document]
Comment (subhashb) [reply src/protean/server/observatory/routes/timeline.py]: Good point. Multi-domain aggregation is the established Observatory pattern (all other route modules iterate domains the same way). In practice, multi-domain setups typically share the same event store or each has its own with non-overlapping global positions. The cursor is best-effort in the multi-domain case — this matches how the existing Observatory pages present aggregated data from multiple domains. If we need strict per-store cursors in the future, we can encode `{domain}:{position}` into the cursor, but that complexity isn't warranted for the current debugging/inspection use case.
